tf2onnx/tfonnx.py

```diff
@@ -1556,6 +1557,9 @@ def fill_op(ctx, node, name, args):
     dtype = value_proto.data_type
     # onnx spec says value MUST be float.
     node.set_attr("value", float(value[0]))
+
+    for i in shape:
+        utils.make_sure(i >= 0, "shape attr should not be less than zero")
```

Comment:
better to move the check higher before actual work starts.

Reply:
done.

my previous thought is that: keep the check before the usage behavior which will hurt you and then when error shows up you can know the reason of error immediate.